src/form/Form.jsx

```diff
@@ -936,6 +944,33 @@ function Form({
           setLoader: () => setButtonLoader(button)
         });
       } else {
+        if (onSkip) {
+          let stepChanged = false;
+          await onSkip({
+            stepName: activeStep.key,
+            userId: initInfo().userKey,
+            setValues: (userVals) => {
+              const values = convertFilesToFilePromises(
+                userVals,
+                fileServarKeys
+              );
+              updateFieldValues(values);
+              client.submitCustom(values);
+            },
+            setStep: (stepKey) => {
+              stepChanged = changeStep(stepKey, activeStep.key, steps, history);
+            },
+            setProgress: (val) => setUserProgress(val),
+            triggerKey: lookupElementKey(
+              activeStep,
+              metadata.elementIDs[0],
+              metadata.elementType
+            ),
+            triggerType: metadata.elementType,
+            triggerAction: metadata.trigger
+          });
+          if (stepChanged) return;
+        }
         handleRedirect({ metadata });
```

Comment:
just want to confirm we still want this handleRedirect to run after the onskip if onskip is true

Reply:
yep! the form will navigate to the next step either way, we additionally optionally run the onSkip callback function if it's passed in